cpp/ql/lib/semmle/code/cpp/ir/implementation/raw/internal/TranslatedStmt.qll

```diff
@@ -734,14 +736,32 @@ class TranslatedTryStmt extends TranslatedStmt {
     // of the `try`, because the exception successor of the `try` itself is
     // the first catch clause.
     handler = this.getHandler(stmt.getNumberOfCatchClauses() - 1) and
-    result = this.getParent().getExceptionSuccessorInstruction(kind)
+    exists(boolean isSEH |
+      stmt instanceof MicrosoftTryStmt and isSEH = true
+      or
+      stmt instanceof TryStmt and isSEH = false
+    |
+      result = this.getParent().getExceptionSuccessorInstruction(kind, isSEH)
+    )
   }
 
-  final override Instruction getExceptionSuccessorInstruction(EdgeKind kind) {
-    result = this.getHandler(0).getFirstInstruction(kind)
-    or
-    not exists(this.getHandler(_)) and
-    result = this.getFinally().getFirstInstruction(kind)
+  final override Instruction getExceptionSuccessorInstruction(EdgeKind kind, boolean isSEH) {
+    // SEH exceptions are only handled for SEH try statements and
+    // C++ exceptions for C++ try statements.
+    // I.e., we are assuming there isn't a mix and match between SEH and C++ exceptions.
+    // They are either all SEH or all C++ within a single try block depending on the
+    // try type (TryStmt vs MicrosoftTryStmt).
+    (
+      stmt instanceof TryStmt and isSEH = false
+      or
+      stmt instanceof MicrosoftTryStmt and isSEH = true
+    ) and
+    (
+      result = this.getHandler(0).getFirstInstruction(kind)
```

Comment:
I think that, if `isSEH` is `true` and `stmt instanceof TryStmt` then we should propagate the exception further up by doing `this.getParent().getExceptionSuccessorInstruction(kind)` and letting some exception handler further up handle it. That removes the assumption that users aren't mixing SEH and C++ exceptions.

Reply:
See above comment about opening a new issue for mix and match exceptions in the future.